### PR TITLE
Show how to request asset price using golang

### PR DESCRIPTION
**Pull request by aelmanaa**
Purpose of this PR is to show how to request Chainlink price feed on any supported EVM chain using golang

### CODE REVIEW

File: pricefeed-golang/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+## Table of Contents
+
+- [About The Project](#about-the-project)
+- [Prerequisites](#prerequisites)
+- [Installation](#installation)
+- [Usage](#usage)
+
+## About The Project
+
+The project shows how to retrieve asset price from any EVM chain. The contract addresses can be found [here](https://docs.chain.link/docs/reference-contracts/).
```
Comment (zeuslawyer) [inline pricefeed-golang/README.md:12]: recommend we make more specific by saying "The <<data feed>> contract addresses.  For beginners it is not obvious what smart contracts are involved.

File: pricefeed-golang/main.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	aggregatorv3 "chainlink-price-feed/aggregatorv3"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func main() {
+
+	err := godotenv.Load(".env")
```
Comment (zeuslawyer) [inline pricefeed-golang/main.go:22]: For consistency, perhaps add comment // Read the .env file

File: pricefeed-golang/main.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	// fetch rpc_url
+	rpcUrl := os.Getenv("RPC_URL")
+	if len(rpcUrl) == 0 {
+		log.Fatal("rpcUrl is empty. check the .env file")
+	}
+
+	// check if chainlink feed proxy is provided in command line
+	args := os.Args[1:]
```
Comment (zeuslawyer) [inline pricefeed-golang/main.go:34]: Can this be shortened to 
if len(os.Args) > 1 && os.Args[1] !==""{
   feedAddress = os.Args[1]
}

File: pricefeed-golang/main.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		feedAddress = os.Getenv("DEFAULT_FEED_ADDR")
+	}
+
+	client, err := ethclient.Dial(rpcUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Price Feed address
+	chainlinkPriceFeedProxyAddress, isContract := isContractAddress(feedAddress, client)
```
Comment (zeuslawyer) [inline pricefeed-golang/main.go:50]: Suggestion:  I think idiomatic Go uses "ok" for booleans that operate as conditional checkers.

See : https://go.dev/doc/effective_go and search for ", ok " on the page.

File: pricefeed-golang/main.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	log.Printf("%v Price feed address is  %v\n", description, chainlinkPriceFeedProxyAddress)
+	log.Printf("Round id is %v\n", roundData.RoundId)
+	log.Printf("Answer is %v\n", roundData.Answer)
+	log.Printf("Formatted answer is %v\n", divide(roundData.Answer, divisor))
+	log.Printf("Started at %v\n", formatTime(roundData.StartedAt))
+	log.Printf("Updated at %v\n", formatTime(roundData.UpdatedAt))
+	log.Printf("Answered in round %v\n", roundData.AnsweredInRound)
+}
+
+func isContractAddress(addr string, client *ethclient.Client) (common.Address, bool) {
```
Comment (zeuslawyer) [inline pricefeed-golang/main.go:87]: This function appears to do two things - validate that it is an address and also converts it to common.Address Type.  making it common.Address is simply one line on line 97. 

So maybe better to make this pure validation, return only 1 value (bool) and simply add the common.HexToAddress after processing the returned boolean?

That way the type conversion step is not hidden in the validation logic and may be easier for people to follow?

**Review by zeuslawyer — COMMENTED** [review]
this is awesome.  so excited to see this go live!

File: pricefeed-golang/README.md (hunk 1 above)
Comment (aelmanaa) [reply pricefeed-golang/README.md:12]: ok good point. will do it . thanks

File: pricefeed-golang/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+1. Get a RPC API Key from a node provider such as [Alchemy](https://www.alchemy.com/), [Infura](https://infura.io/), [Moralis](https://moralis.io/), or [QuickNode](https://www.quicknode.com/). This example uses the KOVAN Ethereum test network.
+1. Clone the repo
+   ```sh
+   git clone https://github.com/smartcontractkit/smart-contract-examples.git
+   ```
+1. Enter the direcory
+   ```sh
+   cd pricefeed-golang
+   ```
+1. Create the `.env` file
```
[the remark this answers is not in the document]
Comment (aelmanaa) [reply pricefeed-golang/README.md:30]: ok good point. will do it . thanks

File: pricefeed-golang/main.go (hunk 2 above)
Comment (aelmanaa) [reply pricefeed-golang/main.go:22]: ok

File: pricefeed-golang/main.go (hunk 5 above)
Comment (aelmanaa) [reply pricefeed-golang/main.go:87]: the outcome of `common.HexToAddress` is used in 2 different places so I didn't want to call it 2 times. But you are right, let's favor simplicity

File: pricefeed-golang/main.go [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+func formatTime(timestamp *big.Int) time.Time {
+	timestampInt64 := timestamp.Int64()
+	if timestampInt64 == 0 {
+		log.Fatalf("timestamp %v cannot be represented as int64", timestamp)
+	}
+	return time.Unix(timestampInt64, 0)
+}
+
+func divide(num1 *big.Int, num2 *big.Int) *big.Float {
```
[the remark this answers is not in the document]
Comment (aelmanaa) [reply pricefeed-golang/main.go:114]: good idea.

**Review by zeuslawyer — APPROVED** [review]
LGTM!

**Pull request merged**